modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,467 @@
+         2. if a column has two children, combined with one of those children
+            having a null dtype.
+         Making the mask concept explicit seems useful. One null dtype would
+         not be enough to cover both bit and byte masks, so that would mean
+         even more checking if we did it the Arrow way.
+    TBD: there's also the "chunk" concept here, which is implicit in Arrow as
+         multiple buffers per array (= column here). Semantically it may make
+         sense to have both: chunks were meant for example for lazy evaluation
+         of data which doesn't fit in memory, while multiple buffers per column
+         could also come from doing a selection operation on a single
+         contiguous buffer.
+         Given these concepts, one would expect chunks to be all of the same
+         size (say a 10,000 row dataframe could have 10 chunks of 1,000 rows),
+         while multiple buffers could have data-dependent lengths. Not an issue
+         in pandas if one column is backed by a single NumPy array, but in
+         Arrow it seems possible.
+         Are multiple chunks *and* multiple buffers per column necessary for
+         the purposes of this interchange protocol, or must producers either
+         reuse the chunk concept for this or copy the data?
+
+    Parameters
+    ----------
+    column : PandasDataframe
+        A ``PandasDataframe`` object.
+    allow_copy : bool, default: True
+        A keyword that defines whether or not the library is allowed
+        to make a copy of the data. For example, copying data would be necessary
+        if a library supports strided buffers, given that this protocol
+        specifies contiguous buffers. Currently, if the flag is set to ``False``
+        and a copy is needed, a ``RuntimeError`` will be raised.
+    offset : int, default: 0
+        The offset of the first element.
+
+    Notes
+    -----
+    This Column object can only be produced by ``__dataframe__``,
+    so doesn't need its own version or ``__column__`` protocol.
+    """
+
+    def __init__(
+        self, column: PandasDataframe, allow_copy: bool = True, offset: int = 0
+    ) -> None:
+        if not isinstance(column, PandasDataframe):
+            raise NotImplementedError(f"Columns of type {type(column)} not handled yet")
+
+        # Store the column as a private attribute
+        self._col = column
+        self._allow_copy = allow_copy
+        self._offset = offset
+
+    @property
+    def size(self) -> int:
+        return len(self._col.index)
+
+    @property
+    def offset(self) -> int:
+        return self._offset
+
+    @property
+    def dtype(self) -> Tuple[DTypeKind, int, str, str]:
+        dtype = self._col.dtypes[0]
+
+        if pandas.api.types.is_categorical_dtype(dtype):
+            return (
+                DTypeKind.CATEGORICAL,
+                32,
+                pandas_dtype_to_arrow_c(np.dtype("int32")),
+                "=",
```

Comment:
[According to the protocol](https://data-apis.org/dataframe-protocol/latest/API.html#:~:text=For%20categoricals%2C%20the%20format%20string%20describes%20the%20type%20of%20the%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20categorical%20in%20the%20data%20buffer.) the type's metadata have to confort with the type of the "data" buffer being returned from `get_buffers()`.

In your implementation, the data buffer contains codes of the category and the codes not always have a `int32` dtype:
```python
>>> pandas.Series(["a", "c", "b"]).astype("category").values.codes
array([0, 2, 1], dtype=int8)
```

Reply:
fixed